sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/DistinctTest.java

```diff
@@ -251,4 +253,64 @@ public void testTriggeredDistinctRepresentativeValues() {
         KV.of(1, "k1"), KV.of(2, "k2"), KV.of(3, "k3"));
     triggeredDistinctRepresentativePipeline.run();
   }
+
+  /**
+   * Regression test: when all values are emitted by a speculative trigger, caused a null KV
+   * when the on-time firing occurred.
+   */
+  @Test
+  @Category({ValidatesRunner.class, UsesTestStream.class})
+  public void testTriggeredDistinctRepresentativeValuesEmpty() {
+    Instant base = new Instant(0);
+    TestStream<KV<Integer, String>> values =
+        TestStream.create(KvCoder.of(VarIntCoder.of(), StringUtf8Coder.of()))
+            .advanceWatermarkTo(base)
+            .addElements(TimestampedValue.of(KV.of(1, "k1"), base))
+            .advanceProcessingTime(Duration.standardMinutes(1))
+            .advanceWatermarkToInfinity();
+
+    PCollection<KV<Integer, String>> distinctValues = triggeredDistinctRepresentativePipeline
+        .apply(values)
+        .apply(Window.<KV<Integer, String>>into(FixedWindows.of(Duration.standardMinutes(1)))
+            .triggering(AfterWatermark.pastEndOfWindow().withEarlyFirings(
+                AfterProcessingTime.pastFirstElementInPane().plusDelayOf(
+                    Duration.standardSeconds(30))))
+            .withAllowedLateness(Duration.ZERO)
+            .discardingFiredPanes())
+        .apply(Distinct.withRepresentativeValueFn(new Keys<Integer>())
+            .withRepresentativeType(TypeDescriptor.of(Integer.class)));
+
+    PAssert.that(distinctValues).containsInAnyOrder(KV.of(1, "k1"));
+    triggeredDistinctRepresentativePipeline.run();
+  }
+
+  /**
+   * Regression test variation: when all values are emitted by a speculative trigger and the values
```

Comment:
What kind of error is this test guarding against? It seems identical except for being weaker than the previous test. If you think it's still good to have both, consider deduplicating the code into a method.

Reply:
I'll delete it. It was really during development to confirm what was and was not the bug. Incidentally, reuse in testing is not usually a good idea vs straight line.

Reply:
Deleted.